sklearn/ensemble/_bagging.py

```diff
@@ -278,10 +278,18 @@ def _fit(self, X, y, max_samples=None, max_depth=None, sample_weight=None):
         random_state = check_random_state(self.random_state)
 
         # Convert data (X is required to be 2d and indexable)
-        X, y = self._validate_data(
-            X, y, accept_sparse=['csr', 'csc'], dtype=None,
-            force_all_finite=False, multi_output=True
-        )
+        from . import IsolationForest  # noqa
+        if isinstance(self, IsolationForest):
```

Comment:
can't you rather modify IsolationForest _get_tags so is_supervised is set to False?

Reply:
It's already False because it inherits from OutlierMixin.

But in the end, the problem isn't there anymore because we just removed one of the checks.